### PR TITLE
Adding common libs needed for api-catalogue and xroad-collector

### PR DESCRIPTION
**Pull request by saevarma**
# Common libs for api-catalogue and xroad-collector

## What

To break down PR #1405  we are creating smallar PR with isolated units.
This contains the libs part that are common for api-catalogue and xroad-collector

## Why

To get api-catalogue and xroad-collector to master but in smaller PRs

## Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] Formatting passes locally with my changes
- [X] I have rebased against master before asking for a review

### CODE REVIEW

File: libs/api-catalogue/elastic/src/lib/types/types.model.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+interface ShardsResponse {
```
Comment (jeremybarbet) [inline libs/api-catalogue/elastic/src/lib/types/types.model.ts:1]: Is this file automatically generated?

Comment (Addips) [reply libs/api-catalogue/elastic/src/lib/types/types.model.ts:1]: No as of now the definitions are not available from elasticsearch, but this based on TypeScript support description here: https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/typescript.html

Comment (jeremybarbet) [reply libs/api-catalogue/elastic/src/lib/types/types.model.ts:1]: I see, then it should be similar to this one used on the api/elastic-search https://github.com/island-is/island.is/blob/master/libs/api/content-search/src/types/index.ts I wonder if we could share some part of this file somewhere instead of typing it locally on each project

Comment (Addips) [reply libs/api-catalogue/elastic/src/lib/types/types.model.ts:1]: Yes, I agree, would be nice to have a lib or something of that sorts for the shared types :)

Comment (saevarma) [reply libs/api-catalogue/elastic/src/lib/types/types.model.ts:1]: So should we maybe create a island-ui/elastic library that other libraries and projects can reuse these basics types?

Comment (jeremybarbet) [reply libs/api-catalogue/elastic/src/lib/types/types.model.ts:1]: Since it's only types, it might be confusing to create a whole lib for it. I wonder if it could be instead inside `libs/island-shared/types/elastic.d.ts` for example

Comment (saevarma) [reply libs/api-catalogue/elastic/src/lib/types/types.model.ts:1]: Sounds good, I will move it there and update the types used in api/content-search as well

Comment (arnifreyrs) [reply libs/api-catalogue/elastic/src/lib/types/types.model.ts:1]: I like that idea @jeremybarbet

Comment (saevarma) [reply libs/api-catalogue/elastic/src/lib/types/types.model.ts:1]: there is a  `libs/shared` with the two libs `form-fields` and `mocking`. Any preference on adding the `types` library there or in a new `libs/island-shared/` ?

Comment (arnifreyrs) [reply libs/api-catalogue/elastic/src/lib/types/types.model.ts:1]: let's use `libs/shared` for this

File: apps/service-portal/src/components/GlobalModules/GlobalModules.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import {
```
Comment (arnifreyrs) [inline apps/service-portal/src/components/GlobalModules/GlobalModules.tsx:1]: is this file intended to be part of this PR? how does service-portal and api-catalogue connect?

**Review by arnifreyrs — COMMENTED** [review]
There are various changes to apps that I am unsure if should be part of this PR

File: apps/service-portal/src/components/GlobalModules/GlobalModules.tsx (hunk 2 above)
Comment (saevarma) [reply apps/service-portal/src/components/GlobalModules/GlobalModules.tsx:1]: I did a `yarn format` to format my changes, admit I reviewed the staged file in a hurry. There is no connection and should no be a part of this PR. I will revert and fix this. Well spotted!
